test/unit/models/document/paginated_timeline_test.rb

```diff
@@ -72,8 +72,7 @@ class PaginatedTimelineTest < ActiveSupport::TestCase
                           submitted
                           updated
                           rejected
-                          submitted
-                          created]
```

Comment:
One of the implicit responsibilities of this test is to prove it correctly differentiates between `"created"` and `"editioned"` actions.

The difference between the two is whether or not the edition is the first on a document – i.e. `@is_first_edition`:

https://github.com/alphagov/whitehall/blob/ea2b5baec0a081d59b4d0b6b5ab463311edd9f49/app/models/document/paginated_history.rb#L54-L57

That gets populated by `PaginatedTimeline#get_versions_hash`:

https://github.com/alphagov/whitehall/blob/ea2b5baec0a081d59b4d0b6b5ab463311edd9f49/app/models/document/paginated_timeline.rb#L145-L147

But by reducing `Document::PaginatedTimeline::PER_PAGE` we lose the initial `"created"` action from this test, so the test becomes less useful.

Instead, could we mock the `PER_PAGE` value in this test so we still get the full list of actions?

We should be able to use the `mock_pagination` helper – something like:

```ruby
mock_pagination(per_page: 30) do
  timeline = Document::PaginatedTimeline.new(document: @document, page: 1)
  entries = timeline.entries.select { |e| e.instance_of?(Document::PaginatedHistory::AuditTrailEntry) }
  expected_actions = %w[updated
                        editioned
                        published
                        submitted
                        updated
                        rejected
                        submitted
                        created]
end
```

Reply:
Cool thanks will sort